### ISSUE TITLE
[Bug] AsyncSubscribe not found

### ISSUE THREAD

**Issue by framctr** [apache/incubator-eventmesh#857]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Environment

Linux

### EventMesh version

1.4.0

### What happened

When invoking `$JAVA $JAVA_OPT -classpath ${DEMO_PATH}/conf:${DEMO_PATH}/apps/*:${DEMO_PATH}/libs/* $DEMO_MAIN` with `DEMO_PATH=org.apache.eventmesh.tcp.demo.sub.eventmeshmessage.AsyncSubscribe`, I get the following error:
`Error: Could not find or load main class org.apache.eventmesh.tcp.demo.sub.eventmeshmessage.AsyncSubscribe`

### How to reproduce

1. Download code and extract as `eventmesh-1.4.0` folder
2. go to `eventmesh-1.4.0/eventmesh-examples` folder and execute `../gradlew build`
3. gradle successfully builds
4. move results to a demo-app folder to keep things ordered (`conf` with `server.env` etc, `apps` with `classes`, and `libs` with jars, slf4j-api and sl4j-simple). If skipping that phase, then the scripts do not work........
5. Launch subscriber script and it fails, but the publisher will work....

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by xwm1992** [comment]
![image](https://user-images.githubusercontent.com/13237619/167761738-3f2ba391-f251-4510-8ec6-c2a987341317.png)
![image](https://user-images.githubusercontent.com/13237619/167761935-37e5f5ed-b0e3-4b0d-a363-ea25f1cf4682.png)

I just follow your steps, but didn't get your ERROR, please take a look

**Comment by framctr** [comment]
Hi, thank you for your response.
I cannot get your results. I will explain better all the steps I do:
```
$ docker run -it --rm openjdk:8 /bin/bash

# cd /opt
# EVENTMESH_VERSION=1.4.0
# wget https://github.com/apache/incubator-eventmesh/archive/refs/tags/v${EVENTMESH_VERSION}.tar.gz -O - | tar -xz
# mv incubator-eventmesh-${EVENTMESH_VERSION} eventmesh-${EVENTMESH_VERSION}
# cd eventmesh-${EVENTMESH_VERSION}/eventmesh-example
# sh /opt/eventmesh-${EVENTMESH_VERSION}/gradlew build
```
At the end of the process, if I go to `eventmesh-examples/bin` and launch `tcp_sub.sh` or 
the publisher, it doesn't work.
```
# cd bin
# ls
http_pub.sh  http_sub.sh  tcp_pub.sh  tcp_pub_broadcast.sh  tcp_sub.sh  tcp_sub_broadcast.sh
# bash tcp_sub.sh
tcp_sub.sh: line 27: warning: setlocale: LC_CTYPE: cannot change locale (en_US.UTF-8): No such file or directory
tcp_sub.sh: line 28: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
tcp_sub_demo use java location= /usr/local/openjdk-8/bin/java
DEMO_HOME : /data/em-example/dist/eventmesh-examples, DEMO_LOG_HOME : /data/em-example/dist/eventmesh-examples/logs
cat: /data/em-example/dist/eventmesh-examples/conf/server.env: No such file or directory
```
I tried to create a similar launch script in this way:
```
# cd /opt/eventmesh-${EVENTMESH_VERSION}/eventmesh-examples
# mkdir -p demo-app/conf
# mkdir -p demo-app/apps
# mkdir -p demo-app/libs
# mv build/resources/main/* demo-app/conf
# mv build/libs/* demo-app/libs
# mv build/classes demo-app/apps
# # Add missing library slf4j
# SLF4J_VERSION=1.7.9
# wget -P demo-app/libs https://repo1.maven.org/maven2/org/slf4j/slf4j-api/${SLF4J_VERSION}/slf4j-api-${SLF4J_VERSION}.jar
# wget -P demo-app/libs https://repo1.maven.org/maven2/org/slf4j/slf4j-simple/${SLF4J_VERSION}/slf4j-simple-${SLF4J_VERSION}.jar
# cd demo-app
# pwd
/opt/eventmesh-1.4.0/eventmesh-examples/demo-app
# ls
apps  conf  libs
```
Now I add in `demo-app` the following scripts (inspired by the ones in the project):
#### tcp_pub.sh
```bash
#!/bin/bash

# This script will launch the demo app as TCP publisher.

# The configuration of different servers may be inconsistent,
# adding these configurations can avoid the problem of garbled characters
# export LANG=en_US.UTF-8
# export LC_CTYPE=en_US.UTF-8
# export LC_ALL=en_US.UTF-8

DEMO_PATH=${1:-$(dirname $0)}

DEMO_LOG_PATH=${DEMO_PATH}/logs
DEMO_LOG=${DEMO_LOG_PATH}/demo_tcp_pub.log

# create logs folder, if already exist does not show err msg
mkdir -p "${DEMO_LOG_PATH}" 2> /dev/null

JAVA="$(which java)"

# Java configuration options
JAVA_OPT=`cat ${DEMO_PATH}/conf/server.env | grep APP_START_JVM_OPTION::: | awk -F ':::' {'print $2'}`
JAVA_OPT="${JAVA_OPT} -XX:+UseG1GC -XX:G1HeapRegionSize=16m -XX:G1ReservePercent=25 -XX:InitiatingHeapOccupancyPercent=30 -XX:SoftRefLRUPolicyMSPerMB=0 -XX:SurvivorRatio=8 -XX:MaxGCPauseMillis=50"
JAVA_OPT="${JAVA_OPT} -verbose:gc -Xloggc:${DEMO_PATH}/logs/demo_tcp_pub_gc_%p.log -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+PrintGCApplicationStoppedTime -XX:+PrintAdaptiveSizePolicy"
JAVA_OPT="${JAVA_OPT} -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=${DEMO_PATH}/logs -XX:ErrorFile=${DEMO_PATH}/logs/hs_err_%p.log"
JAVA_OPT="${JAVA_OPT} -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=5 -XX:GCLogFileSize=30m"
JAVA_OPT="${JAVA_OPT} -XX:-OmitStackTraceInFastThrow"
JAVA_OPT="${JAVA_OPT} -XX:+AlwaysPreTouch"
JAVA_OPT="${JAVA_OPT} -XX:MaxDirectMemorySize=8G"
JAVA_OPT="${JAVA_OPT} -XX:-UseLargePages -XX:-UseBiasedLocking"
JAVA_OPT="${JAVA_OPT} -Dio.netty.leakDetectionLevel=advanced"
JAVA_OPT="${JAVA_OPT} -Dio.netty.allocator.type=pooled"
JAVA_OPT="${JAVA_OPT} -Djava.security.egd=file:/dev/./urandom"
JAVA_OPT="${JAVA_OPT} -Dlog4j.configurationFile=${DEMO_PATH}/conf/log4j2.xml"
#JAVA_OPT="${JAVA_OPT} -Deventmesh.log.home=${DEMO_LOG_PATH}"
JAVA_OPT="${JAVA_OPT} -DconfPath=${DEMO_PATH}/conf"
JAVA_OPT="${JAVA_OPT} -Dlog4j2.AsyncQueueFullPolicy=Discard"
JAVA_OPT="${JAVA_OPT} -Drocketmq.client.logUseSlf4j=true"

DEMO_MAIN=org.apache.eventmesh.tcp.demo.pub.eventmeshmessage.AsyncPublish

# Debug and log
echo "====== tcp publisher demo ======" | tee -a "$DEMO_LOG"
echo "INFO: Starting at $(date)" | tee -a "$DEMO_LOG"
echo "INFO: DEMO_PATH: ${DEMO_PATH}" | tee -a "$DEMO_LOG"
echo "INFO: DEMO_LOG_PATH : ${DEMO_LOG_PATH}" | tee -a "$DEMO_LOG"
echo "INFO: JAVA: $JAVA" | tee -a "$DEMO_LOG"
echo "INFO: DEMO MAIN: $DEMO_MAIN" | tee -a "$DEMO_LOG"

$JAVA $JAVA_OPT -classpath ${DEMO_PATH}/conf:${DEMO_PATH}/apps/*:${DEMO_PATH}/libs/* $DEMO_MAIN | tee -a "$DEMO_LOG"
```

#### tcp_sub.sh
```bash
#!/bin/bash

# This script will launch the demo app as TCP publisher.

# The configuration of different servers may be inconsistent,
# adding these configurations can avoid the problem of garbled characters
# export LANG=en_US.UTF-8
# export LC_CTYPE=en_US.UTF-8
# export LC_ALL=en_US.UTF-8

DEMO_PATH=${1:-$(dirname $0)}

DEMO_LOG_PATH=${DEMO_PATH}/logs
DEMO_LOG=${DEMO_LOG_PATH}/demo_tcp_sub.log

# create logs folder, if already exist does not show err msg
mkdir -p "${DEMO_LOG_PATH}" 2> /dev/null

JAVA="$(which java)"

# Java configuration options
JAVA_OPT=`cat ${DEMO_PATH}/conf/server.env | grep APP_START_JVM_OPTION::: | awk -F ':::' {'print $2'}`
JAVA_OPT="${JAVA_OPT} -XX:+UseG1GC -XX:G1HeapRegionSize=16m -XX:G1ReservePercent=25 -XX:InitiatingHeapOccupancyPercent=30 -XX:SoftRefLRUPolicyMSPerMB=0 -XX:SurvivorRatio=8 -XX:MaxGCPauseMillis=50"
JAVA_OPT="${JAVA_OPT} -verbose:gc -Xloggc:${DEMO_PATH}/logs/demo_tcp_sub_gc_%p.log -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+PrintGCApplicationStoppedTime -XX:+PrintAdaptiveSizePolicy"
JAVA_OPT="${JAVA_OPT} -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=${DEMO_PATH}/logs -XX:ErrorFile=${DEMO_PATH}/logs/hs_err_%p.log"
JAVA_OPT="${JAVA_OPT} -XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=5 -XX:GCLogFileSize=30m"
JAVA_OPT="${JAVA_OPT} -XX:-OmitStackTraceInFastThrow"
JAVA_OPT="${JAVA_OPT} -XX:+AlwaysPreTouch"
JAVA_OPT="${JAVA_OPT} -XX:MaxDirectMemorySize=8G"
JAVA_OPT="${JAVA_OPT} -XX:-UseLargePages -XX:-UseBiasedLocking"
JAVA_OPT="${JAVA_OPT} -Dio.netty.leakDetectionLevel=advanced"
JAVA_OPT="${JAVA_OPT} -Dio.netty.allocator.type=pooled"
JAVA_OPT="${JAVA_OPT} -Djava.security.egd=file:/dev/./urandom"
JAVA_OPT="${JAVA_OPT} -Dlog4j.configurationFile=${DEMO_PATH}/conf/log4j2.xml"
#JAVA_OPT="${JAVA_OPT} -Deventmesh.log.home=${DEMO_LOG_PATH}"
JAVA_OPT="${JAVA_OPT} -DconfPath=${DEMO_PATH}/conf"
JAVA_OPT="${JAVA_OPT} -Dlog4j2.AsyncQueueFullPolicy=Discard"
JAVA_OPT="${JAVA_OPT} -Drocketmq.client.logUseSlf4j=true"

DEMO_MAIN=org.apache.eventmesh.tcp.demo.sub.eventmeshmessage.AsyncSubscribe

# Debug and log
echo "====== tcp subscriber demo ======" | tee -a "$DEMO_LOG"
echo "INFO: Starting at $(date)" | tee -a "$DEMO_LOG"
echo "INFO: DEMO_PATH: ${DEMO_PATH}" | tee -a "$DEMO_LOG"
echo "INFO: DEMO_LOG_PATH : ${DEMO_LOG_PATH}" | tee -a "$DEMO_LOG"
echo "INFO: JAVA: $JAVA" | tee -a "$DEMO_LOG"
echo "INFO: DEMO MAIN: $DEMO_MAIN" | tee -a "$DEMO_LOG"

$JAVA $JAVA_OPT -classpath ${DEMO_PATH}/conf:${DEMO_PATH}/apps/*:${DEMO_PATH}/libs/* $DEMO_MAIN | tee -a "$DEMO_LOG"
```

Now, if I launch them, tcp_pub works, while the subscriber says that `org.apache.eventmesh.tcp.demo.sub.eventmeshmessage.AsyncSubscribe` cannot be found.

**Comment by xwm1992** [comment]
The sh files has been changed, you can use these files under the master branch https://github.com/apache/incubator-eventmesh/tree/master/eventmesh-examples/bin

**Comment by framctr** [comment]
Thank you for your replies.

> The sh files has been changed, you can use these files under the master branch https://github.com/apache/incubator-eventmesh/tree/master/eventmesh-examples/bin

> Or you can build the tar files using the source code, by execute `gradlew clean jar dist && gradlew installPlugin` and `gradlew tar -x test` then you can got the tar.gz file.

Ok I did what you suggested and I used the updated files in `master`.

I noticed that they seem to start, but they are not because of the same error. In the following I inserted just the subscriber script:
```
[root@9c6dc5a83917 bin]# ls
start.sh  stop.sh  tcp_eventmeshmessage_sub.sh  watchdog.sh

[root@9c6dc5a83917 bin]# pwd
/opt/incubator-eventmesh/dist/bin

[root@9c6dc5a83917 bin]# bash tcp_eventmeshmessage_sub.sh
tcp_sub_demo use java location= /usr/local/jdk/bin/java
DEMO_HOME : /opt/incubator-eventmesh/dist, DEMO_LOG_HOME : /opt/incubator-eventmesh/dist/logs
Error: Could not find or load main class org.apache.eventmesh.tcp.demo.sub.eventmeshmessage.AsyncSubscribe
```
I'm on docker and I tried both with openjdk-8 and oracle jdk-8.

It seems to me that gradle does not build correctly the examples (it is successful, but the shell scripts do not start).
I could probably be wrong. In that case, can you provide the steps to make the demo work? Running examples is not well documented.

**Comment by xwm1992** [comment]
OK，sorry I forgot that , we didn't  package the example module into the tar.gz file for docker image. you can find this in the build.gradle file at root project . It doesn't contains the `example module`.
![image](https://user-images.githubusercontent.com/13237619/167825257-62880107-f167-4355-b99b-11e275459835.png)

**Comment by framctr** [comment]
Ok, but I need to test the eventmesh deployment I made on a Kubernetes cluster. Working locally is not what I need.

---

After some testing, I found that gradle builds everything successfully, but the tcp `AsyncSubscribe` and `AsyncSubscribeBroadcast` main methods cannot be found when invoking them.

I'm not a Java developer and I have no experience with gradle and after some research it seems everything ok to me in `build.gradle`.

This does not happen with the Publisher.. could it be related to `implements ReceiveMsgHook<EventMeshMessage>`?

I tried also to rewrite the subscriber demo from the ground up. I've noticed that even in that case the subscriber is not working (no main class can be found)... At this point I think it could be related to the `eventmesh-common` or `eventmesh-sdk` subprojects..
